shell/pages/404.vue

```diff
@@ -0,0 +1,15 @@
+<script>
+import Brand from '@shell/mixins/brand';
+
+export default {
+  mixins: [Brand],
+  beforeMount() {
+    this.$store.commit('setError', { error: new Error('404: This page could not be found') });
+    this.$router.replace('/fail-whale');
+  }
```

Comment:
If route guards don't redirect us we should go to failwhale with the error message we used to emit in the `render()` method of `entry-helpers.js`